rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/ParametricTextDocumentService.java

```diff
@@ -280,6 +298,7 @@ else if (excp != null) {
             }
             logger.trace("Finished parsing tree, reporting new parse error: {} for: {}", newParseError, file.getLocation());
             facts(file.getLocation()).reportParseErrors(file.getLocation(),
+                tree == null ? new Versioned<>(version, null) : tree,
```

Comment:
I don't think we should be putting `null` in `Versioned`. (especially as the type is not `@Nullable`)

Reply:
Yeah, I already didn't like this tbh. This morning I realized that it just doesn't make sense to have a `Versioned<ITree>` to report *parse errors*, as there's no tree to begin with (the version remains relevant, though). So I refactored the code a bit more substantially to be able to make a distinction between reporting of parse errors and reporting of analyze/build errors. The former are now represented as `Versioned<List<Diagnostic>>`, while the latter are represented as `Versioned<WithTree<List<Diagnostic>>>', where `WithTree` is just a small helper class. This replaces the previous `VersionedDiagnostics` class.